File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.java
import java.io.*;
import java.util.*;

class Main {
	public static boolean[] check = new boolean[9];
	public static int[] arr = new int[9];
	
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		for(int i = 0; i < 9; i++){
			arr[i] = Integer.parseInt(br.readLine());
		}
		Arrays.sort(arr);	
		go(0,0);
	}

	public static void go(int st, int idx){
		if(idx == 7){
			if(getSum() == 100){
				for(int i = 0; i < 9; i++){
					if(check[i]) System.out.println(arr[i]);
				}
				System.exit(0);
			}
            return;
		}
		
		for(int i = st; i < 9; i++){
			check[i] = true;
			go(i + 1, idx + 1);
			check[i] = false;
		}
	}
	
	public static int getSum(){
		int sum = 0;
		for(int i = 0; i < 9; i++){
			if(check[i]) sum += arr[i];
		}
		return sum;
	}
}

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include <iostream>
#include <algorithm>
using namespace std;
int main() {
	int arr[9];
	int sum = 0;
	for(int i = 0; i < 9; i++){
		cin >> arr[i];
		sum += arr[i];
	}
	int diff = sum - 100;
	sort(arr, arr + 9);
	int idx1, idx2;
	for(int i = 0; i < 9; i++){
		for(int j = i + 1; j < 9; j++){
			if(arr[i] + arr[j] == diff){
				idx1 = i;
				idx2 = j;
			}
		}
	}

	for(int i = 0; i < 9; i++){
		if(i != idx1 && i != idx2){
			cout << arr[i] << "\n";
		}
	}
	return 0;
}